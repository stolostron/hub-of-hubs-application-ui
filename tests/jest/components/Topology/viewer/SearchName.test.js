// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
"use strict";

import Search from "../../../../../src-web/components/Topology/viewer/SearchName";
import renderer from "react-test-renderer";

const testSearchName = "test";
const locale = "en-US";

describe("SearchName search", () => {
  const props = {
    searchName: testSearchName,
    onNameSearch: jest.fn(),
    locale: locale
  };

  const searchInst = new Search(props);
  searchInst.setNameSearchRef(document);
  searchInst.handleSearch({ value: "test" });
  searchInst.handleClear();
  it("render as expected", () => {
    const component = renderer.create(searchInst.render());
    expect(component.toJSON()).toMatchSnapshot();
  });
});
