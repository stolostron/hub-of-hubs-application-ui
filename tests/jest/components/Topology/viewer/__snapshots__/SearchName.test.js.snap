// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchName search render as expected 1`] = `
<div
  aria-label="Name"
  className="search-filter"
  id="Name"
  role="region"
>
  <div
    className="pf-c-search-input"
    id="search-name"
  >
    <div
      className="pf-c-input-group"
    >
      <div
        className="pf-c-search-input__bar"
      >
        <span
          className="pf-c-search-input__text"
        >
          <span
            className="pf-c-search-input__icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </span>
          <input
            aria-label="Search-input"
            className="pf-c-search-input__text-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Find [+Find]"
            value="test"
          />
        </span>
        <span
          className="pf-c-search-input__utilities"
        >
          <span
            className="pf-c-search-input__clear"
          >
            <button
              aria-disabled={false}
              aria-label="Reset"
              className="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Button-plain-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              disabled={false}
              onClick={[Function]}
              role={null}
              type="button"
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="currentColor"
                height="1em"
                role="img"
                style={
                  Object {
                    "verticalAlign": "-0.125em",
                  }
                }
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
